pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -660,51 +655,70 @@ func (r *ReconcileLogStorage) deleteInvalidECKManagedPublicCertSecret(ctx contex
+	oprKeyCertIssuer, err := utils.GetCertificateIssuer(oprKeyCert.Data[corev1.TLSCertKey])
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
-	// If the provided certificate secret (secret) is managed by the operator we need to check if the secret that
-	// Elasticsearch creates from that given secret (pubSecret) has the expected DNS name. If it doesn't, delete the
-	// public secret so it can get recreated.
-	if pubSecret != nil {
-		operatorManaged, err := utils.IsOperatorManaged(certSecret, "tls.crt")
+	// If Certificate management is enabled, we only want to trust the CA cert andlet the init container handle private key generation.
+	if instl.CertificateManagement != nil {
+		cmCa := instl.CertificateManagement.CACert
+		cmIssuer, err := utils.GetCertificateIssuer(cmCa)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		if cmIssuer != oprKeyCertIssuer && !utils.IsOperatorIssued(oprKeyCertIssuer) {
+			return nil, nil, nil, fmt.Errorf("certificate management does not support custom Elasticsearch secrets, please delete secret %s/%s or disable certificate management", oprKeyCert.Namespace, oprKeyCert.Name)
+		}
+
+		oprKeyCert.Data[corev1.TLSCertKey] = instl.CertificateManagement.CACert
+		esKeyCert = rsecret.CopyToNamespace(render.ElasticsearchNamespace, oprKeyCert)[0]
+		oprCert = createCaCertSecret(instl.CertificateManagement.CACert, relasticsearch.PublicCertSecret, rmeta.OperatorNamespace())
+
+	} else {
+		esKeyCert = rsecret.CopyToNamespace(render.ElasticsearchNamespace, oprKeyCert)[0]
+		// Get the pub secret - might be nil
+		pubSecret, err := utils.GetSecret(ctx, r.client, relasticsearch.PublicCertSecret, render.ElasticsearchNamespace)
 		if err != nil {
 			return nil, nil, nil, err
 		}
 
-		if operatorManaged {
-			err = utils.SecretHasExpectedDNSNames(pubSecret, "tls.crt", svcDNSNames)
-			if err == utils.ErrInvalidCertDNSNames {
-				if err := r.deleteInvalidECKManagedPublicCertSecret(ctx, pubSecret); err != nil {
-					return nil, nil, nil, err
+		operatorManaged := utils.IsOperatorIssued(oprKeyCertIssuer)
+
+		// If the provided certificate secret (secret) is managed by the operator we need to check if the secret that
+		// Elasticsearch creates from that given secret (pubSecret) has the expected DNS name. If it doesn't, delete the
+		// public secret so it can get recreated.
+		if pubSecret != nil {
+			if operatorManaged {
+				err = utils.SecretHasExpectedDNSNames(pubSecret, corev1.TLSCertKey, svcDNSNames)
+				if err == utils.ErrInvalidCertDNSNames {
+					if err := r.deleteInvalidECKManagedPublicCertSecret(ctx, pubSecret); err != nil {
+						return nil, nil, nil, err
+					}
 				}
 			}
-		}
 
-		pubSecret = rsecret.CopyToNamespace(rmeta.OperatorNamespace(), pubSecret)[0]
+			oprCert = rsecret.CopyToNamespace(rmeta.OperatorNamespace(), pubSecret)[0]
+		}
 	}
 
-	return certSecret, certSecretESCopy, pubSecret, nil
+	return oprKeyCert, esKeyCert, oprCert, err
```

Comment:
I think we should rename the third variable to pubSecret. I find having both `oprKeyCert` and `oprCert` confusing.